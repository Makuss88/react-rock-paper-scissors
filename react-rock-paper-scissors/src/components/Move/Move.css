.flex {
	display: flex;
	align-items: center;
	justify-content: space-evenly;
}

.round {
	background: #000;
    padding: 12px 24px;
    border-radius: 100%;
}

.movedIcon {
    font-size: 60px;
    padding-top: 6px;
    display: inline-block;
    vertical-align: middle;
}

.pulsingIcon {
	-webkit-animation: pulse 1s linear infinite;
	-moz-animation: pulse 1s linear infinite;
	-ms-animation: pulse 1s linear infinite;
	animation: pulse 1s linear infinite;
}

@keyframes pulse {
 0% {
    -webkit-transform: scale(1.1);
   	-moz-transform: scale(1.1);
   	-o-transform: scale(1.1);
   	-ms-transform: scale(1.1);
   	transform: scale(1.1);
 }
 50% {
    -webkit-transform: scale(0.8);
   	-moz-transform: scale(0.8);
   	-o-transform: scale(0.8);
   	-ms-transform: scale(0.8);
   	transform: scale(0.8);
 }
 100% {
    -webkit-transform: scale(1);
   	-moz-transform: scale(1);
   	-o-transform: scale(1);
   	-ms-transform: scale(1);
   	transform: scale(1);
 }
}