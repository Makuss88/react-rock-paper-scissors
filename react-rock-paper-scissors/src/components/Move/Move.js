import React, { Component } from 'react';
import './Move.css';

class Move extends Component {
    render() {
        return(
            <div className="round">
                {
                    this.props.selectedOption.icon ?
                    <span className="movedIcon" role="img" aria-label={this.props.selectedOption.text}>
                        { this.props.selectedOption.icon }
                    </span>
                    :
                    <div className="pulsingIcon">
                        <span className="movedIcon" role="img" aria-label="Waiting for the first move">ðŸ‘Š</span>
                    </div>
                }
            </div>
        );
    }
}

export default Move;