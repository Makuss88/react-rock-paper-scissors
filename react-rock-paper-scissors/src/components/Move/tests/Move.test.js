import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Move from '../Move';

Enzyme.configure({ adapter: new Adapter() });

describe('Move component', () => {

    it('renders pulsing icon when user doesn\'t have a selected option', () => {
        const wrapper = shallow(<Move selectedOption={ '' } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    
    it('renders Rock icon when user has selected Rock', () => {
        const mockUserMove = {
            text: 'Rock',
            icon: 'üëä',
          }
        const wrapper = shallow(<Move selectedOption={ mockUserMove } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders Paper icon when user has selected Paper', () => {
        const mockUserMove = {
            text: 'Paper',
            icon: '‚úãÔ∏è',
          }
        const wrapper = shallow(<Move selectedOption={ mockUserMove } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders Scissors icon when user has selected Scissors', () => {
        const mockUserMove = {
            text: 'Scissors',
            icon: '‚úåÔ∏è',
          }
        const wrapper = shallow(<Move selectedOption={ mockUserMove } />); 
        expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  