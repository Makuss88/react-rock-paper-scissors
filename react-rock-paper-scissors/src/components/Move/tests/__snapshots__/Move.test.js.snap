// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move component renders Paper icon when user has selected Paper 1`] = `
<div
  className="round"
>
  <span
    aria-label="Paper"
    className="movedIcon"
    role="img"
  >
    ✋️
  </span>
</div>
`;

exports[`Move component renders Rock icon when user has selected Rock 1`] = `
<div
  className="round"
>
  <span
    aria-label="Rock"
    className="movedIcon"
    role="img"
  >
    👊
  </span>
</div>
`;

exports[`Move component renders Scissors icon when user has selected Scissors 1`] = `
<div
  className="round"
>
  <span
    aria-label="Scissors"
    className="movedIcon"
    role="img"
  >
    ✌️
  </span>
</div>
`;

exports[`Move component renders pulsing icon when user doesn't have a selected option 1`] = `
<div
  className="round"
>
  <div
    className="pulsingIcon"
  >
    <span
      aria-label="Waiting for the first move"
      className="movedIcon"
      role="img"
    >
      👊
    </span>
  </div>
</div>
`;
